cpp/src/copying/contiguous_split.cu

```diff
@@ -16,6 +32,30 @@ namespace detail {
 
 namespace {
 
+// TODO : this is copy of round_up_safe() that removes the exception throwing so that it can be used in a kernel.
+//        where should this ultimately live?
+template <typename S>
+__device__ inline S round_up_safe_nothrow(S number_to_round, S modulus) {
```

Comment:
Well, assuming size_type goes to 64 bit this shouldn't be an issue.  But yes, this could definitely overflow since we could be dealing with super large strings.  Safest thing here might be to just make these fields size_t.

Reply:
Minor note : I think that pow 2 function is missing a - on the modulus:

return (val + (modulus - 1)) & -modulus;